plugin/src/main/java/io/jenkins/plugins/casc/impl/DefaultConfiguratorRegistry.java

```diff
@@ -123,7 +116,7 @@ private Configurator internalLookup(Type type) {
             if (!(actualType instanceof Class)) {
                 throw new IllegalStateException("Can't handle " + type);
             }
-            return lookup(actualType);
+            return internalLookup(actualType); // cache is not reëntrant
```

Comment:
I wonder how often we fall though into this would be called and if it would be a performance issue (or not).
I will on Tuesday if nothing no body beets me to it create a PR with the suggestion from @ben-manes to do a `cache.getIfPresent(actualType)` before falling back to a non caching option.

Reply:
Could do that. Seems like overkill, since I would expect that typically a given type would either always or never be used wrapped in a `Collection`. For example a `CrumbIssuer` is only a singleton in a `GlobalCrumbIssuerConfiguration`, while a `QueueItemAuthenticator` is always a member of a list in `QueueItemAuthenticatorConfiguration`.

Reply:
ok will skip that PR unless others feel different